app/src/components/pages/ScheduleGenerate.tsx

```diff
@@ -1,7 +1,23 @@
+import { Term } from 'components/shared/constants/timetable.constants';
 import React from 'react';
+import { useLocation } from 'react-router-dom';
+
+interface IStateProps {
+  semester: Term;
+  year: Date;
+}
 
 function ScheduleGenerate() {
-  return <div>ScheduleGenerate</div>;
+  const { state } = useLocation();
+  const { semester, year } = state as IStateProps;
+  return (

```

Comment:
File changes for this file can probably be removed as it's added in another PR

Reply:
Fixed it up, thanks for the suggestion